include/deal.II/lac/trilinos_tpetra_vector.h

```diff
@@ -826,18 +947,27 @@ namespace LinearAlgebra
 
       for (size_type i = 0; i < n_elements; ++i)
         {
-          const size_type                         row = indices[i];
-          const TrilinosWrappers::types::int_type local_row =
+          const size_type                   row = indices[i];
+          TrilinosWrappers::types::int_type local_row =
             vector->getMap()->getLocalElement(row);
 
+          Assert(
+            local_row != Teuchos::OrdinalTraits<int>::invalid(),
+            ExcAccessToNonLocalElement(row,
+                                       vector->getMap()->getLocalNumElements(),
+                                       vector->getMap()->getMinLocalIndex(),
+                                       vector->getMap()->getMaxLocalIndex()));
+
           if (local_row != Teuchos::OrdinalTraits<int>::invalid())
             vector_1d(local_row) += values[i];
         }
 
 #  if !DEAL_II_TRILINOS_VERSION_GTE(13, 2, 0)
       vector->template sync<
-        typename Tpetra::Vector<Number, int, types::signed_global_dof_index>::
-          device_type::memory_space>();
+        typename Tpetra::Vector<Number,
+                                int,
+                                types::signed_global_dof_index,
+                                Node>::device_type::memory_space>();
```

Comment:
I don't actually know where the `Node` type here is declared (did you mean to templatize everything on it and it just slipped through because it's not compiled with the `# if` above?) but in any case, as mentioned in the other PR, it would be nice if we could call these types `NodeType`.

Reply:
This just slipped through, so I removed it.